tests/test_connection.py

```diff
@@ -2695,3 +2695,19 @@ def test_htlc_retransmit_order(node_factory, executor):
         assert(result['status'] == 'complete')
 
     # If order was wrong, we'll get a LOG_BROKEN and fixtures will complain.
+
+
+def test_fundchannel_start_alternate(node_factory, executor):
+    ''' Test to see what happens if two nodes start channeling to
+    each other alternately.
+    Issue #4108
+    '''
+    l1, l2 = node_factory.get_nodes(2)
+
+    l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
+
+    l1.rpc.fundchannel_start(l2.info['id'], 100000)
```

Comment:
if you call 'fundchannel' here, instead of `fundchannel_start`, then at some point in the future (soonTM), it'll also test the v2 channel open code.

Reply:
It has to be `fundchannel_start`, since the race condition being tested is when both peers perform `fundchannel_start`.  Otherwise once `fundchannel` returns, the state is different and a different non-buggy part of the code is triggered.

Adding the changes below results in the test passing even without code changes, so the test *has to* use `fundchannel_start`:

```diff
     '''
     l1, l2 = node_factory.get_nodes(2)
+    l1.fundwallet(10**6 + 1000000)
 
     l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
 
-    l1.rpc.fundchannel_start(l2.info['id'], 100000)
+    l1.rpc.fundchannel(l2.info['id'], 100000)
 
     fut = executor.submit(l2.rpc.fundchannel_start, l1.info['id'], 100000)
     with pytest.raises(RpcError):
```